[Denzel Tan Tze En] iP

danzzzerl opened the pull request:
# duke bot
### This bot is mainly a to-do list bot:)
> "Determine when you are the most effective and energized, then plan your tasks within that time." - Sunday Alelaja

This bot is:
- text-based
- easy to learn
- user-friendly
- AND  ~~EASY~~ *SUPER* EASY TO USE

How to use it:
1. Type list to see your current tasks
2. Type todo followed by a todo task to add a todo
3. Type deadline followed by a task and /by followed by a deadline in the format yyyy-mm-dd to add a deadline
4. Type event followed by an event and /at followed by a timing to add a deadline
5. Type mark/unmark followed by the number of the task to mark the task as done or undone
6. Type delete followed by the number of the task to delete the task 
7. Type find followed by a keyword to look for tasks that contain that keyword
8. Type bye to end the bot

and it is **FREE!!** 😎🤩

Features:
- [ ] Managing Tasks
- [ ] Managing Deadlines
- [ ] Managing Events
- [X] Add/Removing tasks and marking them as done/not done
- [ ] Listing out all your current tasks

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```java
public static void main(String[] args) {
        new Duke("./src/main/java/duke.txt").run();
}
```

find the main code at this [Github repo](https://github.com/danzzzerl/ip)

danzzzerl committed:
Level-1

danzzzerl committed:
Level-2

danzzzerl committed:
Level-3

danzzzerl committed:
Level-4

danzzzerl committed:
A-TextUiTesting

danzzzerl committed:
A-TextUiTesting

danzzzerl committed:
Level-5

danzzzerl committed:
Level-6

danzzzerl committed:
Level-7

danzzzerl committed:
Level-7

danzzzerl committed:
Level-8

danzzzerl committed:
Merge branch 'branch-Level-7'

danzzzerl committed:
merged Level-7 and Level-8

danzzzerl committed:
added modified files for Level 7 and 8

danzzzerl committed:
A-MoreOOP

danzzzerl committed:
A-Packages

danzzzerl committed:
A-Gradle

danzzzerl committed:
A-JUnit

danzzzerl committed:
A-JavaDoc

danzzzerl committed:
A-CodingStandard

danzzzerl committed:
Level-9

danzzzerl committed:
Merge branch 'branch-A-JavaDoc'

danzzzerl committed:
merged branch-A-JavaDoc and branch-A-CodingStandard

danzzzerl committed:
Merge branch 'branch-Level-9'

t1mzzz commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
+# duke.Duke project template
```
This might be a result of packaging duke. Should you change this to Duke project template?
```suggestion
# Duke project template
```

t1mzzz commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+3. After that, locate the `src/main/java/duke.Duke.java` file, right-click it, and choose `Run duke.Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
Similar to the above, this might be a result of packaging duke. Should you change duke.Duke to Duke?
```suggestion
3. After that, locate the `src/main/java/duke.Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```

danzzzerl replied:
Oh yes! Thanks!

t1mzzz commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+     */
+    public String deadlineDateStr() {
+        return date.getMonth() + " " + date.getDayOfMonth() + " " + date.getYear();
+    }
```
Should the name of this method be a verb?
```suggestion
    public String getDeadlineDateStr() {
        return date.getMonth() + " " + date.getDayOfMonth() + " " + date.getYear();
    }
```

t1mzzz commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+    public void run() {
+        Ui.sayGreeting();
+        Scanner sc = new Scanner(System.in);
+        boolean canQuit = false;
```
It is excellent practice to name boolean variables this way. Naming boolean variables is a common mistake for many, but you have avoided it well. Well done 👍!

t1mzzz commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+     * The main function.
+     */
+    public static void main(String[] args) {
+        new Duke("./src/main/java/duke.txt").run();
```
Would it be better for you to create a final String variable with this file path?

t1mzzz commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,117 @@
+ * @author Denzel Tan
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
```
Should the naming of this variable be in plural form?

t1mzzz commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,261 @@
+
+            // print message when event is added
+            System.out.println("Got it. I've added this task:\n  " + currTask +
+                    "\nNow you have " + tasks.size() + " tasks in the list.");
```
Perhaps it is better if messages are handled by the Ui?

t1mzzz reviewed (commented):
Overall, your code looks good to me. I found very minor issues in your code, but I may not have found all. I suggest you utilize the Checkstyle file. OOP is used well throughout, but your Parser class might be slightly too long. Do consider splitting these methods into more classes. Other than that, your code looks excellent! Do not forget to add the PR description in this PR. Well done 👍

danzzzerl committed:
Update README.md

Co-authored-by: t1mzzz <97350034+t1mzzz@users.noreply.github.com>

danzzzerl committed:
Update src/main/java/duke/Deadline.java

Co-authored-by: t1mzzz <97350034+t1mzzz@users.noreply.github.com>

danzzzerl committed:
Updated Duke main method to include final String variable FILEPATH

danzzzerl committed:
Updated method name

danzzzerl committed:
Level-10

danzzzerl committed:
branch-Level-10

danzzzerl committed:
branch-A-Varargs

danzzzerl committed:
Merge branch 'branch-Level-10'

danzzzerl committed:
Merge branch 'branch-A-Varargs'

danzzzerl committed:
bye command exits program now

agarwal-anjali commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    /**
+     * Show the date of the deadline in a string.
```
May be you can consider reframing your JavaDoc comments with keywords that are more appropriate in describing the function! For example, here, the function does not 'Show' but 'Returns' the date of the deadline in a string.

agarwal-anjali commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+package duke;
+
+import java.io.*;
```
I feel it is suggested that we must list all imported classes explicitly. So may be you can consider doing that in your future code and avoid using * that imports all classes.

agarwal-anjali commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,31 @@
+
+    private Duke duke = new Duke("./src/main/java/duke.txt");
+
+    @Override
```
For the JavaDoc, perhaps you can consider using the @inheritDoc tag to reuse the header comment from the parent method and also add additional comments if needed.

agarwal-anjali commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,29 @@
+ * @author Denzel Tan
+ */
+public class Todo extends Task {
+    protected String by;
```
Perhaps you can comment or use JavaDocs to explain what is the use of this global variable 'by' as it seems you are not really using this class member within the class!

agarwal-anjali reviewed (commented):
Overall your code looks well structured and I feel you have made good use of OOP principles! Great Work!

danzzzerl committed:
Add Assertions

There are several methods that require certain inputs. Assert helps to make sure these inputs follow the conditions.

danzzzerl committed:
Improve Code Quality

Improved the code quality of certain classes such as Parser.java.

Some methods had too many lines of code and others had arrow-head style code.

Added more methods within the classes to improve OOP and code quality.

danzzzerl committed:
Merge pull request #1 from danzzzerl/branch-A-Assertions

Add Assertions

danzzzerl committed:
Merge branch 'master' into branch-A-CodeQuality

danzzzerl committed:
Merge pull request #2 from danzzzerl/branch-A-CodeQuality

Improve Code Quality

danzzzerl committed:
moved assert code in parse function to the right position

danzzzerl committed:
Add Tagging Extension

danzzzerl committed:
Merge branch 'branch-C-Tagging'

danzzzerl committed:
update readme

danzzzerl committed:
Changed IOException message for loading file
